doc/topics/variables-extending-and-substitutions.rst

```diff
@@ -131,6 +131,91 @@ extends-cache buildout option <extends-cache-buildout-option>`.
 When a relative path is used in an extends option, it's interpreted
 relative to the path of the extending configuration.
 
+Conditional configuration sections
+==================================
+
+Sometimes, you need different configuration in different environments
+(different operating systems, or different versions of Python).  To
+make this easier, you can define environment-specific options by
+providing conditional sections:
+
+.. code-block:: ini
+
+    [ctl]
+    suffix =
+
+    [ctl:windows]
+    suffix = .bat
+
+.. -> conf
+
+    >>> import zc.buildout.configparser
+    >>> import six
+    >>> zc.buildout.configparser.parse(
+    ...     six.StringIO(conf), 'test', lambda : dict(windows=True))
+    {'ctl': {'suffix': '.bat'}}
+    >>> zc.buildout.configparser.parse(
+    ...     six.StringIO(conf), 'test', lambda : dict(windows=False))
+    {'ctl': {'suffix': ''}}
+
+In this tiny example, we've defined a ``ctl:suffix`` option that's
+``.bat`` on Windows and an empty string elsewhere.
+
+A conditional section has a colon and then a Python expression after
+the name.  If the Python expression result is true, the section
+options from the section are included.  If the value is false, the
+section is ignored.
+
+Some things to note:
+
+- If there is no exception, then options from the section are
+  included.
```

Comment:
I don't understand this sentence.  It seems to repeat the previous paragraph somewhat (if the result is true), and contradict it somewhat (if there's no exception, but the value was false, there's no inclusion).  Also, does "exception" here mean a Python exception raised when evaluating the conditional expression?  Maybe it would be better to say what happens if there *is* an exception.